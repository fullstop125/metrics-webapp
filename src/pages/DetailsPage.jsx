import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import { fetchDataBySymbol } from '../redux/companies/companies';

const DetailsPage = () => {
  const { companyDataBySymbol } = useSelector((state) => state.companies);
  const dispatch = useDispatch();
  const { smbl } = useParams();
  useEffect(() => {
    dispatch(fetchDataBySymbol(smbl));
  }, []);
  console.log(smbl);

  console.log(companyDataBySymbol);
  return (
    <>
      <div className='head'>
        <div>
          <img
            src={companyDataBySymbol.image}
            alt='logo'
            style={{ width: 100, height: 100, borderRadius: 50 }}
          />
        </div>
        <div>
          <h3 >{companyDataBySymbol.companyName}</h3>
          <span>({companyDataBySymbol.symbol})</span>
        </div>
      </div>
      <div className='list'>
        <h3 >Company Symbol</h3>
        <p>{companyDataBySymbol.symbol}</p>
      </div>
      <div className='list'>
        <h3>Company Exchange</h3>
        <p>{companyDataBySymbol.exchange}</p>
      </div>
      <div className='list'>
        <h3>Company Industry</h3>
        <p>{companyDataBySymbol.industry}</p>
      </div>
      <div className='list'>
        <h3>Company Website</h3>
        <p>{companyDataBySymbol.website}</p>
      </div>
      <div className='list'>
        <h3>Company CEO</h3>
        <p>{companyDataBySymbol.ceo}</p>
      </div>
      <div className='list'>
        <h3>Company Address</h3>
        <p>{companyDataBySymbol.address}</p>
      </div>
      <div className='list'>
        <h3>Company City</h3>
        <p>{companyDataBySymbol.city}</p>
      </div>
    </>
  );
};

export default DetailsPage;


