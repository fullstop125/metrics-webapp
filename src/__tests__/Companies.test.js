import React from 'react';
import renderer from 'react-test-renderer';
import DetailsPage from '../__Mocks__/Details';

describe('DetailsPage', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<DetailsPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
