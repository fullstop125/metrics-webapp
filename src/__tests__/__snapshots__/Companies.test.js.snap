// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsPage renders correctly 1`] = `
[
  <div
    className="head"
  >
    <div>
      <img
        alt="logo"
        image={true}
        style={
          {
            "borderRadius": 50,
            "height": 100,
            "width": 100,
          }
        }
      />
    </div>
    <div>
      <h3>
        companyName
      </h3>
      <span>
        symbol
      </span>
    </div>
  </div>,
  <div
    className="list"
  >
    <h3>
      Company Symbol
    </h3>
    <p>
      symbol
    </p>
  </div>,
  <div
    className="list"
  >
    <h3>
      Company Exchange
    </h3>
    <p>
      exchange
    </p>
  </div>,
  <div
    className="list"
  >
    <h3>
      Company Industry
    </h3>
    <p>
      industry
    </p>
  </div>,
  <div
    className="list"
  >
    <h3>
      Company Website
    </h3>
    <p>
      website
    </p>
  </div>,
  <div
    className="list"
  >
    <h3>
      Company CEO
    </h3>
    <p>
      ceo
    </p>
  </div>,
  <div
    className="list"
  >
    <h3>
      Company Address
    </h3>
    <p>
      address
    </p>
  </div>,
  <div
    className="list"
  >
    <h3>
      Company City
    </h3>
    <p>
      city
    </p>
  </div>,
]
`;
